fetch/fetch.go

```diff
@@ -437,6 +437,25 @@ func (f *Fetch) send(requests []RequestMessage) {
 	}
 
 	peer2batches := f.organizeRequests(requests)
+	if len(peer2batches) == 0 {
+		f.mu.Lock()
+		defer f.mu.Unlock()
+		for _, msg := range requests {
+			if req, ok := f.ongoing[msg.Hash]; ok {
+				f.logger.WithContext(req.ctx).With().Info("delaying hash request",
+					log.Stringer("hash", msg.Hash),
+					log.String("hint", string(msg.Hint)),
+				)
+				f.unprocessed[msg.Hash] = req
```

Comment:
so GetBlocks will not return immediately with error if no peers, but instead we will do the request when peers are available?

Reply:
yes. essentially put those requests back to the queue

Reply:
this approach is fine for me too, hard to tell what is better

Reply:
actually i prefer returning error right away. it's easier for debugging (and avoiding future bug)